services/QuillLMS/client/app/modules/BigQueryUDFs.js

```diff
@@ -55,55 +94,77 @@ export function studentwiseSkillGroupUDF(scores, activityIds, completedAts, skil
     return row.score
   }
 
+  function deduplicateAndAverageScores(arr) {
```

Comment:
Note how, like previously, `deduplicateAndAverageScores` is defined as an exportable function above, but redefined within the parent UDF here.  This allows unit testing. Not DRY at all. Can be fixed by writing a special-purpose JS bundler, should we continue with this pattern in the future.